tests/test_flags.py

```diff
@@ -0,0 +1,294 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Type
+
+import pytest
+
+from disnake import flags
+
+if TYPE_CHECKING:
+
+    class TestFlags(flags.BaseFlags):
+        """A test class for flag testing."""
+
+        @flags.flag_value
+        def one(self):
+            return 1 << 0
+
+        @flags.flag_value
+        def two(self):
+            return 1 << 1
+
+        @flags.flag_value
+        def four(self):
+            return 1 << 2
+
+
+def test_flag_value_creation() -> None:
+    flag = flags.flag_value(lambda x: 1 << 2)
+    assert 1 << 2 == flag.flag
+
+
+def test_fill_with_flags() -> None:
+    @flags.fill_with_flags()
+    class TestFlags(flags.BaseFlags):
+        """A test class for flag testing."""
+
+        @flags.flag_value
+        def one(self):
+            return 1 << 0
+
+        @flags.flag_value
+        def two(self):
+            return 1 << 1
+
+        @flags.flag_value
+        def four(self):
+            return 1 << 2
+
+    ins = TestFlags()
+
+    assert ins
```

Comment:
```suggestion
    assert TestFlags.VALID_FLAGS = {"one": 1, "two": 2, "four": 4}
```
I feel like this would make more sense as a test